packages/contracts-bedrock/src/libraries/GasPayingToken.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Storage } from "src/libraries/Storage.sol";
+import { Constants } from "src/libraries/Constants.sol";
+import { LibString } from "@solady/utils/LibString.sol";
+
+/// @title GasPayingToken
+/// @notice Handles reading and writing the custom gas token to storage.
+///         To be used in any place where gas token information is read or
+///         written to state. If multiple contracts use this library, the
+///         values in storage should be kept in sync between them.
+library GasPayingToken {
+    /// @notice The storage slot that contains the address and decimals of the gas paying token
+    bytes32 internal constant GAS_PAYING_TOKEN_SLOT = bytes32(uint256(keccak256("opstack.gaspayingtoken")) - 1);
```

Comment:
Is there a reason to store decimals in storage given that it's required to be 18? If not, we can rename this to `GAS_PAYING_TOKEN_ADDR_SLOT` and simplify some of these methods/interfaces

Reply:
Its so that we can easily relax the restriction in the future with very little diff

Reply:
Got it, I'm not up to speed on the motivation for the 18 decimal requirement, but if it may be relaxed I'm good with keeping it

Reply:
18 decimal requirement means that it matches ether (native asset) so UIs can just work